--- vendor/iwlwifi/mvm/quota.c
+++ vendor/iwlwifi/mvm/quota.c
@@ -7,6 +7,7 @@
 #include <net/mac80211.h>
 #include "fw-api.h"
 #include "mvm.h"
+#include "iwchaos_chaos.h"
 
 #define QUOTA_100	IWL_MVM_MAX_QUOTA
 #define QUOTA_LOWLAT_MIN ((QUOTA_100 * IWL_MVM_LOWLAT_QUOTA_MIN_PERCENT) / 100)
@@ -195,10 +196,13 @@
 			 * bindings is in low latency. For this case, allocate
 			 * the minimal required quota for the low latency
 			 * binding.
 			 */
-			qdata->quota = cpu_to_le32(QUOTA_LOWLAT_MIN);
+			qdata->quota = cpu_to_le32(iwchaos_chaos_quota_adjust(
+				i, QUOTA_LOWLAT_MIN));
 		else
 			qdata->quota =
-				cpu_to_le32(quota * data.n_interfaces[i]);
+				cpu_to_le32(iwchaos_chaos_quota_adjust(
+					i, quota * data.n_interfaces[i]));
 
 		WARN_ONCE(le32_to_cpu(qdata->quota) > QUOTA_100,
 			  "Binding=%d, quota=%u > max=%u\n",
