--- vendor/iwlwifi/mvm/coex.c
+++ vendor/iwlwifi/mvm/coex.c
@@ -11,6 +11,7 @@
 #include "iwl-modparams.h"
 #include "mvm.h"
 #include "iwl-debug.h"
+#include "iwchaos_chaos.h"
 
 /* 20MHz / 40MHz below / 40Mhz above*/
 #define BT_CI_20MHZ		0
@@ -594,17 +595,21 @@
 	enum iwl_bt_coex_lut_type lut_type;
 
 	if (mvm->last_bt_notif.ttc_status & BIT(phy_ctxt->id))
-		return LINK_QUAL_AGG_TIME_LIMIT_DEF;
+		return iwchaos_chaos_coex_agg_limit(mvmsta->deflink.sta_id,
+						    LINK_QUAL_AGG_TIME_LIMIT_DEF);
 
 	if (le32_to_cpu(mvm->last_bt_notif.bt_activity_grading) <
 	    BT_HIGH_TRAFFIC)
-		return LINK_QUAL_AGG_TIME_LIMIT_DEF;
+		return iwchaos_chaos_coex_agg_limit(mvmsta->deflink.sta_id,
+						    LINK_QUAL_AGG_TIME_LIMIT_DEF);
 
 	lut_type = iwl_get_coex_type(mvm, mvmsta->vif);
 
 	if (lut_type == BT_COEX_LOOSE_LUT || lut_type == BT_COEX_INVALID_LUT)
-		return LINK_QUAL_AGG_TIME_LIMIT_DEF;
+		return iwchaos_chaos_coex_agg_limit(mvmsta->deflink.sta_id,
+						    LINK_QUAL_AGG_TIME_LIMIT_DEF);
 
 	/* tight coex, high bt traffic, reduce AGG time limit */
-	return LINK_QUAL_AGG_TIME_LIMIT_BT_ACT;
+	return iwchaos_chaos_coex_agg_limit(mvmsta->deflink.sta_id,
+					    LINK_QUAL_AGG_TIME_LIMIT_BT_ACT);
 }
